src/newmatrix.cpp

```diff
@@ -3110,6 +3065,75 @@ void matrix_update()
   }
 }
 
+// For things that should happen no more than once an hour.
+void matrix_hour_update()
+{
+  PERF_PROF_SCOPE(pr_, __func__);
+  rnum_t rnum = 1;
```

Comment:
Since this isn't used outside of the for-loop, please define it in the loop initializer (`for (rnum_t rnum = 1; ...)`)

Reply:
Yeah, this was a copy/paste from the existing `matrix_update()`, didn't really think about it.

Reply:
Done